Src/GoogleApis.Auth.DotNet4/OAuth2/ServiceAccountCredential.cs

```diff
@@ -72,6 +76,16 @@ public Initializer(string id, string tokenServerUrl) : base(tokenServerUrl)
                 Scopes = new List<string>();
             }
 
+            /// <summary>Extracts a <see cref="Key"/> from the clientCredentialParameters which is obtained from parsing the Json Key.</summary>
+            public Initializer FromJsonKey(ClientCredentialParameters clientCredentialParameters)

```

Comment:
By the way, you are only using the Pkcs8PrivateKey key, why not just passing this method the key itself, why do we need all the other parameters?